src/Microsoft.DotNet.Tools.Repl.Csi/Program.cs

```diff
@@ -19,21 +24,128 @@ public static int Main(string[] args)
+            }
+
+            return context;
+        }
+
+        private static CommandResult CompileProject(ProjectContext projectContext, string configuration, out string tempOutputDir)
+        {
+            tempOutputDir = Path.Combine(projectContext.ProjectDirectory, "bin", ".dotnetrepl", Guid.NewGuid().ToString("N"));
+
+            Reporter.Output.WriteLine($"Compiling {projectContext.RootProject.Identity.Name.Yellow()} for {projectContext.TargetFramework.DotNetFrameworkName.Yellow()} to use with the {"C# REPL".Yellow()} environment.");
+
+            return Command.Create($"dotnet-compile", $"--output \"{tempOutputDir}\" --temp-output \"{tempOutputDir}\" --framework \"{projectContext.TargetFramework}\" --configuration \"{configuration}\" {projectContext.ProjectFile.ProjectDirectory}")
+                                .ForwardStdOut(onlyIfVerbose: true)
+                                .ForwardStdErr()
+                                .Execute();
+        }
+
+        private static List<string> GetRuntimeDependencies(ProjectContext projectContext, string buildConfiguration)
+        {
+            var runtimeDependencies = new List<string>();
+
+            var projectExporter = projectContext.CreateExporter(buildConfiguration);
+            var projectDependencies = projectExporter.GetDependencies();
+
+            foreach (var projectDependency in projectDependencies)
+            {
+                var runtimeAssemblies = projectDependency.RuntimeAssemblies;
+
+                foreach (var runtimeAssembly in runtimeAssemblies)
+                {
+                    var runtimeAssemblyPath = runtimeAssembly.ResolvedPath;
+
+                    if (!runtimeDependencies.Contains(runtimeAssemblyPath))
+                    {
+                        runtimeDependencies.Add(runtimeAssemblyPath);
+                    }
+                }
+            }
+
+            return runtimeDependencies;
+        }
+
+        private static string CreateResponseFile(ProjectContext projectContext, string buildConfiguration, string tempOutputDir)
+        {
+            var outputFileName = projectContext.ProjectFile.Name;
+            var outputFilePath = Path.Combine(tempOutputDir, $"{outputFileName}{Constants.DynamicLibSuffix}");
+            var projectResponseFilePath = Path.Combine(tempOutputDir, $"dotnet-repl.{outputFileName}{Constants.ResponseFileSuffix}");
+
+            var runtimeDependencies = GetRuntimeDependencies(projectContext, buildConfiguration);
+
+            using (var fileStream = new FileStream(projectResponseFilePath, FileMode.Create))
+            {
+                var streamWriter = new StreamWriter(fileStream);
+
+                streamWriter.WriteLine($"/r:\"{outputFilePath}\"");
+
+                foreach (var projectDependency in runtimeDependencies)
+                {
+                    streamWriter.WriteLine($"/r:\"{projectDependency}\"");
+                }
+
+                streamWriter.Flush();
```

Comment:
Fixed. Sadly, CoreCLR StreamWriter doesn't seem to support this constructor: https://msdn.microsoft.com/en-us/library/fysy0a4b(v=vs.110).aspx.